### PR TITLE
hotfix/awiesm compile on mistral

### LINKED ISSUE #158: Shared Library Error on DKRZ using ECHAM-6.3.05p2-Concurrent Radioan
After successfully compiling `AWIESM-2.1` on DKRZ using the [`hotfix/dkrz_awiesm_2_1`](https://github.com/esm-tools/esm_tools/pull/153) branch for ESM tools, I attempted a small run. However, the model immediately fails with a shared library error. 

```
/mnt/lustre02/work/ab0246/a270077/For_Aprita/experiments/PI_test_001/run_20000101-20001231/work/./echam6: error while loading shared libraries: libpnetcdf.so: cannot open shared object file: No such file or directory
```

### PR DESCRIPTION
**Pull request by pgierz**
# Hotfix: `AWIESM-2.1` Compiles on Mistral

This merge request allows AWIESM to be compiled on MISTRAL. This worked previously, but somehow broke in the meantime. It *might* require changes to `mistral.yaml`, thus I want to put this online to inform other developers (@seb-wahl for GEOMAR models, @denizural for ICON, @mandresm for MPIESM)

Once we have figured out what exactly needs to change to get AWIESM running again, and if that change affects mistral.yaml, I'd ask you to please test your models before we merge this.....

## To be tested (Compile):

**AWIESM + Standalone Version**
+ [x] AWI-ESM-2.1 on Ollie
+ [x] ECHAM-6.3.05-concurrent-rad on Ollie
+ [x] AWI-ESM-2.1 on Mistral
+ [x] ECHAM-6.3.05-concurrent-rad on Mistral

**AWICM (Standard Version)**
+ [x] AWICM-2.0 on Mistral

**MPIESM**
+ [x] MPIESM compiles in mistral in all its versions

**ICON**
@denizural : Please add whatever you want to test

**GEOMAR Models**
@seb-wahl: Please add whatever you want to test

## How to test:
```
cp ${HOME}/.esmtoolsrc ${HOME}/.esmtoolsrc_backup
cd /wherever/you/like
python -m venv .venv
source .venv/bin/activate
git clone https://github.com/esm-tools/esm_tools.git
cd esm_tools
git checkout hotfix/dkrz_awiesm_2_1
./install.sh
esm_versions check
echo "Please proceed with your tests"
echo "Do not forget to restore the esmtoolsrc backup!!"
```

### CODE REVIEW

File: configs/setups/awiesm/awiesm-2.2.yaml [hunk 1]
```diff
@@ -71,6 +71,13 @@ echam:
                         nproca: 24
                         nprocb: 18
 
+        add_export_vars:
```
Comment (mandresm) [inline configs/setups/awiesm/awiesm-2.2.yaml]: Do this actually work out of a `<...>_environment_changes` block?

**Review by mandresm — APPROVED** [review]
Appart from the question above about the `add_export_vars` out of an `<...>_environment_changes` block, I would like to ask if the many ` configs/machines/mistral.yaml.*` files are necessary?

- [x]  MPIESM compiles in mistral in all it's versions.

**Comment by pgierz** [comment]
No, I will remove them. They were just for testing purposes.

File: configs/setups/awiesm/awiesm-2.2.yaml (hunk 1 above)
Comment (pgierz) [reply configs/setups/awiesm/awiesm-2.2.yaml]: ...it does not

**Pull request closed without merging**